docs/orchestrating-elastic-stack-applications/security/saml-authentication.asciidoc

```diff
@@ -0,0 +1,134 @@
+ifdef::env-github[]
+****
+link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-{page_id}.html[View this document on the Elastic website]
+****
+endif::[]
+[id="{p}-{page_id}"]
+= SAML Authentication
+
+The Elastic Stack supports SAML single sign-on (SSO) into Kibana, using Elasticsearch as a backend service. 
+
+NOTE: Elastic Stack SSO requires a valid Enterprise license or Enterprise trial license. See <<{p}-licensing,the license documentation>> for more details about managing licenses.
+
+TIP: Make sure you read the complete link:https://www.elastic.co/guide/en/elasticsearch/reference/current/saml-guide-stack.html[Configuring SAML single-sign-on on the Elastic Stack] guide before  setting up SAML SSO for Kibana and Elasticsearch deployments managed by ECK.
+
+== Add a SAML realm to X-Pack security settings 
+
+To enable SAML SSO for the Elastic Stack, you have to configure the SAML realm in Elasticsearch and enable the usage of the SAML realm and authentication provider in Kibana.
+
+=== Elasticsearch
+
+To add the SAML realm to Elasticsearch, use the `spec` section of the manifest. The SAML realm configuration contains an `idp.metadata.path` field that should be set to the path where your IdP’s SAML metadata file is located in the Elasticsearch pods.
+
+NOTE: The `sp.*` SAML settings must point to Kibana endpoints that are accessible from the web browser used to open Kibana.
+
+Check Elastic link:https://www.elastic.co/guide/en/elasticsearch/reference/current/saml-guide-stack.html#saml-guide-idp[Sack SAML documentation] for more information on `idp.*` and `sp.*` settings.
+
+Make sure not to disable Elasticsearch’s file realm set by ECK, as ECK relies on the file realm for its operation. Set the `order` setting of the SAML realm to a greater value than the `order` value set for the file and native realms, which is by default -100 and -99 respectively. We recommend setting the priority of SAML realms to be lower than other realms, as shown in the next example.
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: elasticsearch.k8s.elastic.co/{eck_crd_version}
+kind: Elasticsearch
+metadata:
+  name: elasticsearch-sample
+spec:
+  version: {version}
+  nodeSets:
+  - name: default
+    count: 1
+    config:
+      xpack.security.authc.realms:
+        saml:
+          saml1:
+            attributes.principal: nameid
+            idp.entity_id: https://sso.example.com/
+            idp.metadata.path: /usr/share/elasticsearch/config/saml/idp-saml-metadata.xml
+            order: 2
+            sp.acs: https://kibana.example.com/api/security/v1/saml
+            sp.entity_id: https://kibana.example.com
+            sp.logout: https://kibana.example.com/logout
+----
+
+The `idp.metadata.path` setting should point to your Identity Provider’s metadata file. To make the file available to Elasticsearch, you need to create a Kubernetes secret containing the metadata file and mounting it to the Elasticsearch container.
```

Comment:
This can also be a URL in which case users can skip the K8s Secret step iirc.

Reply:
Yes tho this is not mentioned in the Elasticsearch docs and I didn't want it to be inconsistent.

Reply:
OK

Reply:
URL support is actually mentioned [here](https://www.elastic.co/guide/en/elasticsearch/reference/7.x/security-settings.html#ref-saml-settings). I'll mention it is supported.